syntax = "proto3";

package fraud_detection;

service HelloService {
    rpc SayHello (HelloRequest) returns (HelloResponse);
}

service FraudDetectionService {
    rpc CheckFraud (FraudDetectionRequest) returns (FraudDetectionResponse);
}

message HelloRequest {
    string name = 1;
}

message HelloResponse {
    string greeting = 1;
}

message FraudDetectionRequest {
    User user = 1;
    CreditCard credit_card = 2;
    string user_comment = 3;
    Address billing_address = 4;
}

message FraudDetectionResponse {
    bool is_fraudulent = 1;
}

message User {
    string name = 1;
    string email = 2;
}

message CreditCard {
    string number = 1;
    string expiration_date = 2;
    string cvv = 3;
}

message Address {
    string street = 1;
    string city = 2;
    string state = 3;
    string zip = 4;
    string country = 5;
}